### PR TITLE
[X86] Print 'l' section flag for SHF_X86_64_LARGE

### PR DESCRIPTION
**Pull request by aeubanks**
When directly compiling to an object file we properly set the section flag, but not when emitting assembly.

### CODE REVIEW

File: llvm/test/CodeGen/X86/code-model-elf-sections.ll [hunk 1]
```diff
@@ -16,6 +16,8 @@
 ; RUN: llc < %s -relocation-model=pic -filetype=obj -code-model=large -data-sections -o %t
 ; RUN: llvm-readelf -S %t | FileCheck %s --check-prefix=SMALL-DS
 
+; RUN: llc < %s -relocation-model=pic -filetype=asm -code-model=medium -large-data-threshold=79 -o - | FileCheck %s --check-prefix=LARGE-ASM
```
Comment (rnk) [inline llvm/test/CodeGen/X86/code-model-elf-sections.ll:19]: This is an MC code change, so the best place to test this is in MC itself. I think you can simply tests that it roundtrips. This assembly code doesn't repro the 'l' flag:
`  .section .lrodata,"al"`
--> `llvm-mc t.s -o -` -->
`        .section        .lrodata,"a",@progbits`

Comment (MaskRay) [reply llvm/test/CodeGen/X86/code-model-elf-sections.ll:19]: We can reuse `llvm/test/MC/ELF/section.s`

**Review by rnk — APPROVED** [review]
Thanks!

**Review by RKSimon — APPROVED** [review]
LGTM, cheers

**Pull request merged**